packages/plugins/type-designer/src/components/drawer-content.svelte

```diff
@@ -0,0 +1,86 @@
+<script lang="ts">
+// CONSTANTS
+import { SCD_ELEMENTS } from '@oscd-plugins/core'
+// SMUI COMPONENTS
+import Textfield from '@smui/textfield'
+import Button from '@smui/button'
+// STORES
+import { pluginStore } from '@/stores/plugin.store'
+import { drawerStore } from '@oscd-plugins/ui'
+// TYPES
+import type { DataTypeTemplates } from '@oscd-plugins/core'
+
+//====== INITIALIZATION ====//
+
+// props
+export let typeElement: DataTypeTemplates.TypeElement
+export let currentColumn: keyof typeof SCD_ELEMENTS
+
+let localAttributes: Record<(typeof standardAttributes)[number], string>
+
+//====== REACTIVITY ====//
+$: ({ element, ...attributes } = typeElement)
+$: standardAttributes = SCD_ELEMENTS[currentColumn]?.element.standardAttributes
+$: setNewLocalAttributes(attributes)
+
+//====== FUNCTIONS =====//
+
+function updateTypeElement() {
+	pluginStore.createAndDispatchUpdateActionEvent({
+		element,
+		oldAttributes: attributes,
+		newAttributes: localAttributes
+	})
+	drawerStore.handleCloseDrawer()
+}
+
+function setNewLocalAttributes(
+	newAttributes: Record<(typeof standardAttributes)[number], string>
+) {
+	localAttributes = { ...localAttributes, ...newAttributes }
+}
+</script>
+
+<div id="type-designer-drawer">
+	<div class="form">
```

Comment:
Do we want to have a system in place, to prevent the form from "blank-submissions"?

Reply:
That should be discuss with the community. But the files can be changed by hand anyway, so we might integrate some sanitizing and keep our forms free from potential XSS attacks.